lib/streamaggr/streamaggr.go

```diff
@@ -708,83 +726,101 @@ func (a *aggregator) runFlusher(pushFunc PushFunc, alignFlushToInterval, skipInc
 		}
 	}
 
-	if dedupInterval <= 0 {
-		alignedSleep(interval)
-		t := time.NewTicker(interval)
-		defer t.Stop()
+	flushDeadline := time.Now().Truncate(interval).Add(interval)
+	tickInterval := time.Duration(a.tickInterval) * time.Millisecond
+	alignedSleep(tickInterval)
+
+	var dedupIdx, flushIdx int
 
-		if alignFlushToInterval && skipIncompleteFlush {
-			a.flush(nil, interval, true)
-			ignoreFirstIntervals--
+	t := time.NewTicker(tickInterval)
+	defer t.Stop()
+
+	isSkippedFirstFlush := false
+	for tickerWait(t) {
+		ct := time.Now()
+
+		dedupTime := ct.Truncate(tickInterval)
+		if a.ignoreOldSamples {
+			dedupIdx, flushIdx = a.getFlushIndices(dedupInterval, interval, dedupTime, flushDeadline)
 		}
```

Comment:
It seems to me like aggregation windows show use a separate flag to be enabled.
Currently, it is not obvious why & how to enable aggregation windows. It would be great to add a separate flag and update the documentation to clarify how this works, when it can be useful and how it affects performance/resource usage.

Reply:
windowing requires to use minTimestamp, which is also used by a logic, enabled by ignoreOldSamples. Agree with a need to have a documentation and partially agree with a separate flag